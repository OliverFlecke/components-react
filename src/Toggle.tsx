import React from 'react';

interface ToggleProps extends React.InputHTMLAttributes<HTMLInputElement> {}

const Toggle = ({ ...checkboxProps }: ToggleProps) => {
	return (
		<label className="switch relative inline-block w-16 h-9">
			<input type="checkbox" {...checkboxProps} />
			<span className="slider"></span>
		</label>
	);
};

export default Toggle;
