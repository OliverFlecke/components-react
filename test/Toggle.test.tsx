import React from 'react';
import Toggle from '../src/Toggle';
import renderer from 'react-test-renderer';

describe('Toggle', () => {
	test('Render snapshot', () => {
		const tree = renderer.create(<Toggle />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
