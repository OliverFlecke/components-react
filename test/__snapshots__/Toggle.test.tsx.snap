// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Render snapshot 1`] = `
<label
  className="switch relative inline-block w-16 h-9"
>
  <input
    type="checkbox"
  />
  <span
    className="slider"
  />
</label>
`;
